### PR TITLE
Allow to go through clippy lints page without javascript

### LINKED ISSUE #13536: website: dark color them renders code unreadable
### Description

https://rust-lang.github.io/rust-clippy/master/index.html#/panic

select the "coal" theme for example

Image

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #13536.

This is the follow-up of https://github.com/rust-lang/rust-clippy/pull/13269.

This PR makes it possible to expand/collapse lints (individually) without JS. To achieve this result, there are two ways:
1. Use `details` and `summary` tags. Problem with this approach is that the web browser search may open the `details` tags automatically if content matching it is inside. From a previous discussion with @Alexendoo, it seems to not be a desired behaviour.
2. Use a little trick where you use a `label` and a checkbox where the checkbox is in fact hidden. Then it's just a matter of CSS.

r? @Alexendoo 

changelog: Allow to go through clippy lints page without JS

### CODE REVIEW

File: util/gh-pages/script.js [hunk 1]
```diff
@@ -138,18 +138,20 @@ function onEachLazy(lazyArray, func) {
...
-    onEachLazy(elem.querySelectorAll("pre > code.language-rust:not(.highlighted)"), el => {
+function highlightIfNeeded(lintId) {
+    const elem = document.getElementById(lintId);
+    const call = el => {
         hljs.highlightElement(el.parentElement)
         el.classList.add("highlighted");
-    });
+    };
+    onEachLazy(elem.querySelectorAll("pre > code.language-rust:not(.highlighted)"), call);
+    onEachLazy(elem.querySelectorAll("pre > code.language-no_run:not(.highlighted)"), call);
```
Comment (Alexendoo) [inline util/gh-pages/script.js:148]: Ah this should not be happening, some changes from #13359 may need to be reapplied on top of those for #13269

Comment (GuillaumeGomez) [reply util/gh-pages/script.js:148]: Oh I see. Gonna update then.

**Comment by GuillaumeGomez** [comment]
Fixed the bug. I took an approach a little bit different by using the `pulldown-cmark` API. Maybe we should do the same in `sanitize_explanation` (in another PR)?

**Comment by Alexendoo** [comment]
It should use `sanitize_explanation` to ensure they're consistent, e.g. this impl is missing the hidden line removal

**Comment by GuillaumeGomez** [comment]
Changed the code to use `sanitize_explanation`. :+1:

**Review by Alexendoo — COMMENTED** [review]
Looks good with the minus sign change, the configuration defaults are unaligned but if I merge with master that's fixed so looks unrelated to this change

**Comment by GuillaumeGomez** [comment]
Fixed the minus sign.

**Review by Alexendoo — APPROVED** [review]

**Comment by Alexendoo** [comment]
👍

@bors r+

**Pull request merged**